glean-core/csharp/Glean/Glean.cs

```diff
@@ -47,6 +47,11 @@ public sealed class Glean
         /// </summary>
         private const string LogTag = "glean/Glean";
 
+        /// <summary>
+        /// This is the name of the platform this instance is running on.
+        /// </summary>
+        private readonly static string PlatformName = $"C# on {Environment.OSVersion}";
```

Comment:
I think this should really be `Environment.OSVersion.Platform`, otherwise you'll get a version number in this (e.g. "Microsoft Windows NT 6.2.9200.0" vs "Win32NT").